tests/test_util.py

```diff
@@ -222,5 +223,14 @@ async def test_wrong_content_type(self):
             )
 
 
+class TestCheckNewline(unittest.TestCase):
+    def test_newline(self):
+        fp = open("tests/com.jetbrains.PhpStorm.json", "r+", encoding="utf-8")
+        self.assertEqual(_check_newline(fp), True)
```

Comment:
Ideally, we should do a full "newline preservation" test, i.e. create a dummy `.json` file with newline, read it, dump it, check that the newline is there, and do the same for a file without trailing newline.

Reply:
If I'm reading this correctly, should I add new json files specifically for this test rather than reuse the ones I did?

And the test should also check whether the json file is correctly generated with the trailing newline, rather than relying only on the existing test at https://github.com/flathub/flatpak-external-data-checker/blob/master/tests/test_checker.py#L141?

Reply:
> should I add new json files specifically for this test rather than reuse the ones I did?

Yes, although you can embed the files contents into the test code instead of storing as real files in git.

>And the test should also check whether the json file is correctly generated with the trailing newline

Not necessary, but probably it would be even better.